### ISSUE TITLE
Configs for Image Classification (cifar10)

### ISSUE THREAD

**Issue by keroro824** [google-research/long-range-arena#16]
Thanks for the great work!
I have a question regarding the hyperparams for training cifar10. I used the setting in this repo and replaced several hyperparams (eg n_layers n_heads etc) with the ones reported in the paper, but the best testing acc I got was 0.36:

> import ml_collections
> 
> NUM_EPOCHS = 200
> TRAIN_EXAMPLES = 45000
> VALID_EXAMPLES = 10000
> 
> 
> def get_config():
>   """Get the default hyperparameter configuration."""
>   config = ml_collections.ConfigDict()
>   config.batch_size = 256
>   config.eval_frequency = TRAIN_EXAMPLES // config.batch_size
>   config.num_train_steps = (TRAIN_EXAMPLES // config.batch_size) * NUM_EPOCHS
>   config.num_eval_steps = VALID_EXAMPLES // config.batch_size
>   config.weight_decay = 0.
>   config.grad_clip_norm = None
> 
>   config.save_checkpoints = True
>   config.restore_checkpoints = False
>   config.checkpoint_freq = (TRAIN_EXAMPLES //
>                             config.batch_size) * NUM_EPOCHS // 2
>   config.random_seed = 0
> 
>   config.learning_rate = .0005 (if using 0.01 from the paper, the loss is not going down)
>   config.factors = 'constant * linear_warmup * cosine_decay'
>   config.warmup = (TRAIN_EXAMPLES // config.batch_size) * 1
>   config.steps_per_cycle = (TRAIN_EXAMPLES // config.batch_size) * NUM_EPOCHS
> 
>   # model params
>   config.model = ml_collections.ConfigDict()
>   config.model.emb_dim = 32
>   config.model.num_heads = 4
>   config.model.num_layers = 3
>   config.model.qkv_dim = 64
>   config.model.mlp_dim = 128
>   config.model.dropout_rate = 0.3
>   config.model.attention_dropout_rate = 0.2
>   config.model.classifier_pool = 'CLS'
>   config.model.learn_pos_emb = True
> 
>   config.trial = 0  # dummy for repeated runs.
>   return config

Could you point out which params I could adjust to match the accuracy (this is for full attention).

**Comment by vanzytay** [comment]
@MostafaDehghani for clarity on image tasks.

Note: we might take awhile to reply due to the upcoming deadlines. Thanks

**Comment by MostafaDehghani** [comment]
Thank you @keroro824 for the question. 

So if I understood correctly, you are looking for the configs for the vanilla transformer to reproduce its results on CIFAR10 dataset in LRA. For that, you can use the followings as the model hparams:

```
  # model params
  config.model = ml_collections.ConfigDict()
  config.model.emb_dim = 128
  config.model.num_heads = 8
  config.model.num_layers = 1
  config.model.qkv_dim = 64
  config.model.mlp_dim = 128
  config.model.dropout_rate = 0.3
  config.model.attention_dropout_rate = 0.2
  config.model.classifier_pool = 'CLS'
  config.model.learn_pos_emb = True
  ```
  
We are planning to release the code for all models and the best performing configurations, as soon as possible. In the meantime, please let us know if you had any questions :)

**Comment by keroro824** [comment]
@MostafaDehghani Thank you !!! I can replicate it now!

**Comment by MostafaDehghani** [comment]
No problem at all! Perfect! 
... and good luck with the deadline :)
